python/pyspark/sql/connect/session.py

```diff
@@ -338,9 +342,35 @@ def catalog(self) -> "Catalog":
 
     catalog.__doc__ = PySparkSession.catalog.__doc__
 
+    def __del__(self) -> None:
+        try:
+            # Try its best to close.
+            self.client.close()
+        except Exception:
+            pass
+
     def stop(self) -> None:
+        # Stopping the session will only stop the current session by closing the connection,
+        # whereas the regular PySpark session closes the Spark Context itself, meaning that
+        # closing all Spark sessions. It is controversial to follow the existing
+        # the regular Spark session's behavior specifically in Spark Connect
+        # the Spark Connect server is designed for multi-tenancy - the remote client side
+        # cannot just stop the server and close other remote clients being used from other
+        # users.
```

Comment:
I believe this is not accurate for existing behavior. 

Closing the connection won't stop the session immediately but after 1 hour inactivity. In one example that a user close the client then re-attach again immediately with the same user id, the second connection will re-use the previous session.

I suggestion we document to match existing behavior.

Meanwhile we have plan to update the mechanism.

Reply:
I assume this comment is a general comment that not limited to local mode? If this is actually only for local mode because the below can stop the session then this comment becomes make sense.

Reply:
Yeah this is general comment. Let me correct this. Thanks for pointing it out.